pkg/gadgets/trace/open/tracer/bpf/opensnoop.bpf.c

```diff
@@ -95,35 +105,38 @@ int ig_openat_e(struct trace_event_raw_sys_enter* ctx)
 static __always_inline
 int trace_exit(struct trace_event_raw_sys_exit* ctx)
 {
-	struct event event = {};
-	struct args_t *ap;
+	struct event *event;
 	int ret;
 	u32 pid = bpf_get_current_pid_tgid();
 	u64 uid_gid = bpf_get_current_uid_gid();
 	u64 mntns_id;
 
-	ap = bpf_map_lookup_elem(&start, &pid);
-	if (!ap)
-		return 0;	/* missed entry */
+	event = bpf_map_lookup_elem(&start, &pid);
+	if (!event)
+		return 0; /* missed entry */
+
 	ret = ctx->ret;
 	if (targ_failed && ret >= 0)
 		goto cleanup;	/* want failed only */
 
 	/* event data */
-	event.pid = bpf_get_current_pid_tgid() >> 32;
-	event.uid = (u32) uid_gid;
-	event.gid = (u32) (uid_gid >> 32);
-	bpf_get_current_comm(&event.comm, sizeof(event.comm));
-	bpf_probe_read_user_str(&event.fname, sizeof(event.fname), ap->fname);
-	event.flags = ap->flags;
-	event.mode = ap->mode;
-	event.ret = ret;
-	event.mntns_id = gadget_get_mntns_id();
-	event.timestamp = bpf_ktime_get_boot_ns();
+	event->pid = bpf_get_current_pid_tgid() >> 32;
+	event->uid = (u32) uid_gid;
+	event->gid = (u32) (uid_gid >> 32);
+	bpf_get_current_comm(&event->comm, sizeof(event->comm));
+	event->ret = ret;
+	event->mntns_id = gadget_get_mntns_id();
+	event->timestamp = bpf_ktime_get_boot_ns();
+
+	// Attempting to extract the full file path with symlink resolution
+	if (0 <= ret)
```

Comment:
I know some people prefer to have the constant first in a comparison, but all our code uses the other way around. Please swap them.

Reply:
OK